src/librustc/middle/dead.rs

```diff
@@ -259,8 +259,9 @@ impl<'a, 'tcx> Visitor<'tcx> for MarkSymbolVisitor<'a, 'tcx> {
     }
 
     fn visit_arm(&mut self, arm: &'tcx hir::Arm) {
-        if arm.pats.len() == 1 {
-            let variants = arm.pats[0].necessary_variants();
+        let pats = arm.top_pats_hack();
```

Comment:
I didn't understand how the logic here works wrt. avoiding or-patterns.

Reply:
This seems to have been added in https://github.com/rust-lang/rust/pull/25961 by @sanxiyn 4 years ago.

Reply:
Thanks, worked perfectly! Fixed.